lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -1111,6 +1127,36 @@ void AtenLenTOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
   });
 }
 
+//===----------------------------------------------------------------------===//
+// AtenMulLeftTOp
+//===----------------------------------------------------------------------===//
+
+void AtenMulLeftTOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
+                                                 MLIRContext *context) {
+  // `[3] * 5` -> `[3,3,3,3,3]`, if it is not mutated.
+  patterns.add(+[](AtenMulLeftTOp op, PatternRewriter &rewriter) {
+    auto listLiteral = op.getL().getDefiningOp<Torch::PrimListConstructOp>();
+    if (!listLiteral || isListPotentiallyMutated(listLiteral))
+      return failure();
+
+    int64_t numReps;
+    if (!matchPattern(op.getN(), m_TorchConstantInt(&numReps)))
+      return failure();
+
+    SmallVector<Value> newListElements;
+    for (int rep = 0; rep < numReps; ++rep) {
+      for (auto operand : listLiteral.getOperands()) {
+        newListElements.push_back(operand);
+      }
+    }
+
+    rewriter.replaceOpWithNewOp<PrimListConstructOp>(
+        op, Torch::ListType::get(newListElements[0].getType()),
```

Comment:
I think it would be more reliable to do 
```suggestion
        op, op.getL().getType(),
```
E.g., when the list has heterogeneous tensor types, the list type will become more specific to the first element: `!torch.list<vtensor>` might become `!torch.list<vtensor<[1,2,3],f32>>` which causes a few issues, both with the updating of types for subsequent consumer ops, and for not being a valid type for other elements of the list with different shapes.

Reply:
Actually, better yet `op.getType()` will be the most reliable, since this op also returns a list.

Reply:
Thanks, I didn't think about lists with heterogeneous types.
Changed in last commit.